src/sklearn_evaluation/plot/pca.py

```diff
@@ -0,0 +1,102 @@
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from sklearn_evaluation.telemetry import SKLearnEvaluationLogger
+from ploomber_core.exceptions import modify_exceptions
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def _set_ax_settings(ax, targets=None):
+    ax.set_title('Principal Component Plot', fontsize=10)
+    ax.set_xlabel('First principal component', fontsize=10)
+    ax.set_ylabel('Second principal component', fontsize=10)
+    if targets is not None:
+        ax.legend(targets)
+    ax.grid()
+
+
+@modify_exceptions
+@SKLearnEvaluationLogger.log(feature='plot')
+def pca(X, Y=None, target_names=None, n_components=2, colors=None, ax=None):
+    """
+    Plot principle component analysis curve.
+
+    Parameters
+    ----------
+
+    X : array-like, shape = [n_samples, n_features]
+        Training data, where n_samples is the number of samples and
+        n_features is the number of features
+
+    Y : array-like or list or None
+        set None if Ignored otherwise, pass the targets here
+
+    n_components : int, float or 'mle', default=None
+        Number of components to keep.
+
+    ax : matplotlib Axes
+        Axes object to draw the plot onto, otherwise uses current Axes
+
+    Notes
+    -----
+    .. versionadded:: 0.10.0
+
+    Returns
+    -------
+    ax: matplotlib Axes
+        Axes containing the plot
+
+    Examples
+    --------
+    .. plot:: ../examples/pca.py
+
+    """
+    if np.isnan(X).sum():
+        raise ValueError('X array should not consist nan '
+                         'pca')
+
+    # Standardizing the features
+    X = StandardScaler().fit_transform(X)
+
+    # PCA Projection to 2D
+    _pca(X, Y, target_names, n_components, colors, ax)
+
+    return ax
+
+
+def _pca(X, Y=None, target_names=None, n_components=2, colors=None, ax=None):
+
+    pca = PCA(n_components=n_components)
+    principalComponents = pca.fit_transform(X)
+
+    if ax is None:
+        _, ax = plt.subplots()
+
+    targets = None
+    if Y is None:
+        ax.scatter(principalComponents[:, 0], principalComponents[:, 1], s=50)
+    else:
+        Y = Y if isinstance(Y, np.ndarray) else np.array(Y)
+
+        if len(Y.shape) < 2:
+            Y = np.expand_dims(Y, 1)
+        new_principalComponents = np.hstack((principalComponents, Y))
+
+        from random import randint
+        targets = np.unique(new_principalComponents[..., -1])
+        colors = ['#%06X' % randint(0, 0xFFFFFF)
```

Comment:
I think we can export this logic to a helper function and reuse it later in other plots (to make coloring and styling consistent). something like `get_default_colors(targets)` (we don't need to export it yet, but refactoring will be easier in the feature).

Reply:
Agreed. But I think we can open an issue on this for now and work on it once more elaborate specifications are listed down (since it should work for all plots)